lib/nautic_net/packet_handler/set_time_from_gps.ex

```diff
@@ -13,9 +14,22 @@ defmodule NauticNet.PacketHandler.SetTimeFromGPS do
 
   @impl NauticNet.PacketHandler
   def handle_packet(%Packet{parameters: %GNSSPositionDataParams{datetime: gps_datetime = %DateTime{}}}, _config) do
+    maybe_set_system_clock(gps_datetime)
+  end
+
+  def handle_packet(_packet, _config), do: :ok
+
+  @impl NauticNet.PacketHandler
+  def handle_data(%Data{values: %NMEA.DateTimeParams{datetime: gps_datetime = %DateTime{}} = data}, _config) do
```

Comment:
I think `values` is a list, right?

```suggestion
  def handle_data(%Data{values: [%NMEA.DateTimeParams{datetime: gps_datetime = %DateTime{}} = data}], _config) do
```